foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/databaseaccess/DatasourceCall.java

```diff
@@ -484,6 +485,25 @@ public void translateCustomQuery() {
         }
     }
 
+    /**
+     * INTERNAL:
+     * Remove line comments from query string.
+     * @param queryString
+     * @return Query string without line comments.
+     */
+    private String removeLineComments(String queryString) {
+        StringTokenizer multiLineTokenizer = new StringTokenizer(queryString, "\n");
+        StringBuffer sb = new StringBuffer();
+
+        while (multiLineTokenizer.hasMoreTokens())  {
+            String s = multiLineTokenizer.nextToken();
+            if (!"--".equals(s.trim().substring(0,2)))
```

Comment:
What about comments at the end of the line?

```sql
insert into table_name (col1, col2)
  values
     (1,2), -- a comment is also possible here, right?
     (3,4);
```

Reply:
Also `--` can be part of a string, e.g.

```sql
-- valid comment ? (1)
insert into table_name (textCol)
  values -- valid comment ? (2)
     ('test test -- no comment ?
-- no comment ? test test
test test');
```
Above code would remove comment (1), not detect comment (2) and accidentally remove the second "no comment" line inside a string.